keycache/keycache.go

```diff
@@ -133,7 +133,11 @@ func (cache *Cache) MatchUser(name, user string, labels []string) (ActiveUser, s
 func (cache *Cache) useKey(name, user, slot string, labels []string) {
 	if val, slot, present := cache.MatchUser(name, user, labels); present {
 		val.Usage.Uses -= 1
-		cache.setUser(val, name, slot)
+		if val.Usage.Uses <= 0 {
+			delete(cache.UserKeys, DelegateIndex{name, slot})
```

Comment:
useKey is not called until/unless the decryption operation succeeds in DecryptKey or DecryptShares, so this change shouldn't affect the behavior when these fail. 

Either way, I think it is correct to delete the delegation as soon as Uses reaches zero, rather than later when we call Refresh().